server/routes/overview/index.ts

```diff
@@ -57,5 +58,8 @@ export default (router: Router, services: Services) => {
     asyncMiddleware(viewArchivedGoalsController.viewArchivedGoals),
   ])
 
-  router.get('/plan/:prisonNumber/view/goals', [asyncMiddleware(viewGoalsController.viewGoals)])
+  router.get('/plan/:prisonNumber/view/goals', [
+    retrieveAllGoalsForPrisoner(services.educationAndWorkPlanService),
```

Comment:
This is correct 👍 
What we are doing here is calling the middleware as part of the route definition
The middleware runs first, and puts the prisoner goals on res.locals 👍 
And then the controller method is invoked.

Which means .... your controller can just assume that the goals are already on res.locals, so you dont need to inject the service into the controller, and you dont need to call the service to lookup the goals within the controller